service/controller/v10/credential/credential.go

```diff
@@ -0,0 +1,67 @@
+package credential
+
+import (
+	"encoding/json"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/api/core/v1"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/aws-operator/service/controller/v10/key"
+)
+
+const (
+	roleKey = "aws-operator"
+)
+
+type role struct {
+	ARN string `json:"arn"`
+}
+
+func GetRole(k8sClient kubernetes.Interface, obj interface{}) (*role, error) {
+	credential, err := readCredential(k8sClient, obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	role, err := getRole(credential)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return role, nil
+}
+
+func readCredential(k8sClient kubernetes.Interface, obj interface{}) (*v1.Secret, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	credentialName := key.CredentialName(customObject)
+	credentialNamespace := key.CredentialNamespace(customObject)
+
+	credential, err := k8sClient.CoreV1().Secrets(credentialNamespace).Get(credentialName, apismetav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return credential, nil
+}
+
+func getRole(credential *v1.Secret) (*role, error) {
+	var r role
+
+	data, ok := credential.Data[roleKey]
+	if !ok {
+		return nil, microerror.Maskf(roleNotFound, roleKey)
+	}
+
+	err := json.Unmarshal(data, &r)
```

Comment:
Oh wait am I wrong here? 

I get `map[string][]byte` from credential.Data, but is that json ?

Reply:
This should work, but I think types wrong. Check https://github.com/giantswarm/giantswarm/issues/3100 for the secret format. `roleKey` should be `aws`, and role type doesn't reflect the format under `aws` key.

Reply:
done in #933 > https://github.com/giantswarm/aws-operator/pull/933/commits/52a462e67364282a325114bf0532e4c8a9412240